source/FreeRTOS_IPv6.c

```diff
@@ -55,6 +64,63 @@ const struct xIPv6_Address FreeRTOS_in6addr_any = { 0 };
  */
 const struct xIPv6_Address FreeRTOS_in6addr_loopback = { { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 } };
 
+/**
+ * This variable is initialized by the system to contain the unspecified IPv6 address.
+ */
+static const struct xIPv6_Address xIPv6UnspecifiedAddress = { { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 } };
+
+/*
+ * Check if the packet is a legal loopback packet.
+ */
+static BaseType_t xIsIPv6Loopback( const IPHeader_IPv6_t * const pxIPv6Header );
+
+/**
+ * @brief Get the group ID and stored into IPv6_Address_t.
+ *
+ * @param[in] pxIPv6Address: The multicast address to filter group ID.
+ *
+ * @return IPv6_Address_t with group ID only.
+ */
+static IPv6_Address_t xGetIPv6MulticastGroupID( const IPv6_Address_t * pxIPv6Address )
+{
+    IPv6_Address_t xReturnGroupID = { 0 };
+
+    configASSERT( pxIPv6Address != NULL );
+
+    ( void ) memcpy( &( xReturnGroupID.ucBytes[ 2 ] ), &( pxIPv6Address->ucBytes[ 2 ] ), 14 );
+
+    return xReturnGroupID;
+}
+
+/**
+ * @brief Check if the packet is a legal loopback packet.
```

Comment:
I think, we do not need legal , we can just say "Check if the packet is a loopback packet." Same in line 100.